### PR TITLE
Oscar now displays the full names of users on the "Orders" page

### PR DESCRIPTION
**Pull request by Nickersoft**
This merge also requires a merge on ecommerce-api-client and edx-platform with tyler-jwt-fullname, so keep that in mind.

@clintonb @rlucioni FYI: @wedaly @jimabramson

### CODE REVIEW

File: ecommerce/settings/local.py [hunk 1]
```diff
@@ -92,12 +92,12 @@
 
 # AUTHENTICATION
 # Set these to the correct values for your OAuth2/OpenID Connect provider (e.g., devstack)
-SOCIAL_AUTH_EDX_OIDC_KEY = 'replace-me'
```
Comment (clintonb) [inline ecommerce/settings/local.py:95]: This file should not be committed.

File: ecommerce/user/migrations/0004_auto_20150514_1427.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (clintonb) [inline ecommerce/user/migrations/0004_auto_20150514_1427.py]: You should only need one migration. Django is weird in that it will create migrations for minor changes, like changing  the display name. You can fix this by deleting both migrations, and running `./manage.py make migrations` again.

**Comment by Nickersoft** [comment]
Updated the PR @clintonb @rlucioni

**Comment by clintonb** [comment]
Minor: Update your commit message to be more aligned with the actual changes you made:
- Added full_name field to model
- Updated JWT auth to set user's full name from token

Otherwise, this is :+1:. Please wait for a second approval before merging.

File: ecommerce/user/models.py [hunk 3]
```diff
@@ -1,9 +1,14 @@
 from django.contrib.auth.models import AbstractUser
 from jsonfield.fields import JSONField
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
 
 
 class User(AbstractUser):
     """Custom user model for use with OIDC."""
+
+    full_name = models.CharField(_('full name'), max_length=255, blank=True, null=True)
```
Comment (rlucioni) [inline ecommerce/user/models.py]: Minor: the verbose name you've given this field (`'full name'`) is the same as the name Django would automatically assign the field. To stay consistent with the convention we've established when giving models and fields verbose names, please give this field the verbose name `'Full Name'`.

File: ecommerce/user/models.py [hunk 4]
```diff
@@ -16,3 +21,6 @@ def access_token(self):
     class Meta(object):
         get_latest_by = 'date_joined'
         db_table = 'ecommerce_user'
+
+    def get_full_name(self):
+        return self.full_name or super(User, self).get_full_name()
```
Comment (rlucioni) [inline ecommerce/user/models.py:26]: What's the benefit of calling `AbstractUser`'s [`get_full_name()`](https://github.com/django/django/blob/stable/1.7.x/django/contrib/auth/models.py#L394-L399) here? If `full_name` is `None`, this would ensure that the empty string is returned, since our `User`'s `first_name` and `last_name` are always blank. You could achieve the same effect more clearly with:

``` python
return self.full_name or ''
```

File: ecommerce/extensions/api/tests/test_authentication.py [hunk 5]
```diff
@@ -101,3 +101,16 @@ def test_authenticate(self):
...
         self.assertEqual(self.auth.authenticate(request), (user, self.DEFAULT_TOKEN))
+
+
+class JwtAuthenticationTests(TestCase):
+    def test_authenticate_credentials_user_creation(self):
+        """ Test whether the user model is being assigned fields from the payload. """
+        full_name = 'George Costanza'
+        email = 'gcostanza@gmail.com'
+        username = 'gcostanza'
+        payload = {"username": username, 'email': email, 'full_name': full_name}
```
Comment (rlucioni) [inline ecommerce/extensions/api/tests/test_authentication.py]: Nitpick: For the sake of code cleanliness, try to avoid mixing quote levels like this. As a rule of thumb, use single quotes unless the text being quoted will be displayed to a user (e.g., as copy or a log message).

File: ecommerce/extensions/api/tests/test_authentication.py [hunk 6]
```diff
@@ -101,3 +101,16 @@ def test_authenticate(self):
 
         request = self._create_request()
         self.assertEqual(self.auth.authenticate(request), (user, self.DEFAULT_TOKEN))
+
+
+class JwtAuthenticationTests(TestCase):
+    def test_authenticate_credentials_user_creation(self):
+        """ Test whether the user model is being assigned fields from the payload. """
+        full_name = 'George Costanza'
```
Comment (rlucioni) [inline ecommerce/extensions/api/tests/test_authentication.py]: For a more robust test, use a Unicode string here. I often generate Unicode for testing purposes using [this tool](http://www.panix.com/~eli/unicode/convert.cgi).

File: ecommerce/user/models.py (hunk 4 above)
Comment (Nickersoft) [inline ecommerce/user/models.py:26]: This was the advice of @clintonb. He said that if a user was using the old system, calling the super method would be required to keep the code from breaking. However, if the first and last name are always empty, I have no problem with making this change. Just want to make sure @clintonb agrees with this.

Comment (clintonb) [inline ecommerce/user/models.py:26]: The `first_name` and `last_name` fields may not always be blank. Calling the super method accounts for this case.

Comment (Nickersoft) [inline ecommerce/user/models.py:26]: Okay, so then I'll leave it as is

**Comment by rlucioni** [comment]
Once the commits in this PR are squashed and tests pass, :+1: .

File: ecommerce/extensions/api/tests/test_authentication.py [hunk 7]
```diff
@@ -101,3 +102,16 @@ def test_authenticate(self):
 
         request = self._create_request()
         self.assertEqual(self.auth.authenticate(request), (user, self.DEFAULT_TOKEN))
+
+
+class JwtAuthenticationTests(TestCase):
+    def test_authenticate_credentials_user_creation(self):
+        """ Test whether the user model is being assigned fields from the payload. """
+        full_name = 'Ｇｅｏｒｇｅ Ｃｏｓｔａｎｚａ'
```
Comment (jimabramson) [inline ecommerce/extensions/api/tests/test_authentication.py:110]: :-)

**Comment by jimabramson** [comment]
I like the code in this PR but have a question on the strategy.  Is the plan to keep first name, last name, and full name around indefinitely?   If we instead add the full name column and backpopulate `first_name + ' ' + last_name` into it, we can remove the need to check in two places.

**Comment by Nickersoft** [comment]
@jimabramson That would be a question for @rlucioni or @clintonb

**Comment by clintonb** [comment]
@Nickersoft please update the commit message.

**Comment by Nickersoft** [comment]
@clintonb Can that just be done by committing and forcing a push?

**Comment by clintonb** [comment]
@jimabramson Splitting is difficult due to the fact that different locales have different definitions of first and last name, which is why these should really be family and given name fields. Ultimately, this should all be done by the LMS since it is the source of record for user accounts.

As for the lookup, keeping the empty fields doesn't hurt us, aside from a negligible loss of disk space. SQL query complexity should not be affected in any significant manner.

**Comment by clintonb** [comment]
@Nickersoft https://help.github.com/articles/changing-a-commit-message/

**Comment by Nickersoft** [comment]
@clintonb Changed the message

**Comment by clintonb** [comment]
@Nickersoft something has gone awry. You have five commits, many of which aren't actually yours.

**Comment by Nickersoft** [comment]
@clintonb I see that.. Not sure why. How do you propose I fix that? Should I just delete the commits and rebase?
